resharper/resharper-unity/src/Yaml/Psi/Caches/UnityEditorPropertyValues/SceneHierarchy.cs

```diff
@@ -0,0 +1,268 @@
+
+            var id = new FileID(null, anchor);
+            var correspondingId = GetCorrespondingSourceObjectId(referenceValues);
+            var prefabInstanceId = GetPrefabInstanceId(referenceValues);
+            bool isStripped = simpleValues.ContainsKey("stripped");
+            if (referenceValues.ContainsKey(UnityYamlConstants.FatherProperty))
+            { 
+                // transform component
+                var rootOrder = int.TryParse(simpleValues.GetValueSafe(UnityYamlConstants.RootOrderProperty), out var r)
+                    ? r
+                    : -1;
+
+                var gameObject = referenceValues.GetValueSafe(UnityYamlConstants.GameObjectProperty);
+                var father = referenceValues.GetValueSafe(UnityYamlConstants.FatherProperty);
+                
+                Elements.Add(id, new TransformHierarchyElement(id, correspondingId, 
+                    prefabInstanceId, isStripped, rootOrder, gameObject, father));
+
+                if (Elements.TryGetValue(gameObject, out var element))
+                {
+                    var goElement = (element as GameObjectHierarchyElement).NotNull("goElement != null");
+                    goElement.TransformId = id;
+                }
+                else
+                {
+                    myGameObjectsTransforms[gameObject] = id;
+                }
+            }
+            else if (referenceValues.ContainsKey(UnityYamlConstants.GameObjectProperty))
+            {
+                // component
+                var gameObject = referenceValues.GetValueSafe(UnityYamlConstants.GameObjectProperty);
+                Elements.Add(id, new ComponentHierarchyElement(id, correspondingId, prefabInstanceId, gameObject, isStripped));
+            }
+            else
+            {
+                // gameobject
+                var transformId = myGameObjectsTransforms.GetValueSafe(id);
+                if (transformId != null)
+                    myGameObjectsTransforms.Remove(transformId);
+                
+                Elements.Add(id, new GameObjectHierarchyElement(id, correspondingId, prefabInstanceId, isStripped, transformId,
+                    simpleValues.GetValueSafe(UnityYamlConstants.NameProperty)));
+            }
+        }
+
+        private FileID GetCorrespondingSourceObjectId(Dictionary<string, FileID> referenceValues)
+        {
+            return referenceValues.GetValueSafe(UnityYamlConstants.CorrespondingSourceObjectProperty) ??
+                   referenceValues.GetValueSafe(UnityYamlConstants.CorrespondingSourceObjectProperty2017);
+        }
+        
+        private FileID GetPrefabInstanceId(Dictionary<string, FileID> referenceValues)
+        {
+            return referenceValues.GetValueSafe(UnityYamlConstants.PrefabInstanceProperty) ??
+                   referenceValues.GetValueSafe(UnityYamlConstants.PrefabInstanceProperty2017);
+        }
+
+        public void AddPrefabModification(IBuffer buffer)
```

Comment:
I will pray for this algorithm, cause I don't really understand what's going on here. 🕯